utxo/src/cache.rs

```diff
@@ -351,16 +176,12 @@ impl<'a> UtxosCache<'a> {
         // self.memory_usage must be deducted from this entry's size
 
         // check whether this entry is fresh
-        if entry.is_fresh {
+        if entry.is_fresh() {
             // This is only available in this view. Remove immediately.
             self.utxos.remove(outpoint);
         } else {
             // mark this as 'spent'
-            let new_entry = UtxoEntry {
-                status: UtxoStatus::Spent,
-                is_dirty: true,
-                is_fresh: false,
-            };
+            let new_entry = UtxoEntry::new(None, false, true);
```

Comment:
I don't like these booleans - they are easy to misuse. Will do something about them

Reply:
Flags are not good either. They're OK for storage, but they're not OK for function calls and interface operations. We need to find a solution that's type-safe. In fact, two enums or some combination structs with good overrides may solve this problem. Please think of something.

Reply:
Enums were also in my mind, but I thought they are too verbose. Though now I think it might be an optimal solution.

Will go with enums